src/shared/sol-conffile.c

```diff
@@ -948,26 +1135,50 @@ _bsearch_entry_cb(const void *data1, const void *data2)
     return _entry_sort_cb(data1, *p);
 }
 
+static const char *
+get_real_type(const char *alias)
+{
+    uint16_t i;
+    struct sol_alias_ctx *ctx;
+
+    SOL_VECTOR_FOREACH_IDX (&node_aliases_map, ctx, i) {
+        char *aux_alias;
+        uint16_t j;
+
+        SOL_PTR_VECTOR_FOREACH_IDX (&ctx->aliases, aux_alias, j) {
+            if (streq(aux_alias, alias))
+                return ctx->node;
+        }
+    }
+
+    return NULL;
+}
+
 static int
 _resolve_config(const char *id, const char **type, const char ***opts)
 {
     struct sol_conffile_entry key;
     struct sol_conffile_entry *entry;
     void **vector_pointer;
+    const char *real_type;
+
+    real_type = get_real_type(id);
```

Comment:
I'd do this as a last-try, not first. Then you only pay the price of the lookup when everything else fails.